students/chris_lombardi/Lesson04/trigrams.py

```diff
@@ -0,0 +1,70 @@
+#UWPCE, PY210
+#Lesson04, Trigrams
+
+import random
+import string
+
+def build_trigram(words):
+    """Build a trigram association from a list of words and return
+    the trigram association as a dictionary"""
+    trigrams = {}
+
+    for i in range(len(words) - 2):
+        pair = words[i:i + 2]
+        trunc_pair = [remove_punctuation(pair[0]), remove_punctuation(pair[1])]
+        follower = words[i + 2]
+        trunc_follower = remove_punctuation(follower)
+        key_pair = (trunc_pair[0], trunc_pair[1])
+
+        trigrams.setdefault(key_pair,[])
+        trigrams[key_pair].append(trunc_follower)
+
+    return trigrams
+
+def trigram_text(trigrams, length_string):
+    """Create a new string based on the trigram association."""
+    new_text = []
+    first_pair = random.choice(list(trigrams.keys()))
+    new_text.extend(first_pair)
+    avail_words = trigrams.get(first_pair)
+    new_text.append(random.choice(trigrams.get(first_pair)))
+
+    for i in range(1,length_string-1):
+        next_pair = tuple(new_text[i:i+2])
+        #Test for a valid key in the trigram dictionary.
+        if trigrams.get(next_pair) == None:
```

Comment:
you would use `is` to compare objects (None is an object); for example `is None` `is True`

in this case however, you can just do `if trigrams.get(next_pair):`